### ISSUE TITLE
Fix SSA aliasing in StableHLOToBatchNormTrainingOpConversionPattern that silently   corrupted every torch.nn.LayerNorm on tt-xla

### ISSUE THREAD

**Issue by saiarthiraguram** [tenstorrent/tt-mlir#8561]
### Describe the bug

  torch.nn.LayerNorm on tt-xla silently produces output_std ≈ input_std² instead of normalized output. Invisible to unit-variance
  Gaussian tests (1² = 1) but catastrophic for transformer activations whose stddev is in the tens-to-thousands.
  
  Verified affected: MapTR Nano R18 110e end-to-end PCC = 0.0519 pre-fix (per-stage outputs all degraded; pts_bbox_head decoder logits
   clamp/saturate from the inflated post-LayerNorm magnitudes).

  Scope by mechanism (not empirically tested beyond MapTR):
  
  high for a single LayerNorm because magnitudes scale uniformly; failure becomes visible once multiple LayerNorms stack or the
  inflated values flow into ops that overflow.
  - Does not fire for torch.compile(backend="tt") (Dynamo decomposes LayerNorm into primitive ops before StableHLO emission).
  - Does not fire for torch.nn.BatchNorm1d/2d callers (they pass real, distinct weight/bias parameters; no CSE aliasing).
  
 ### Steps to reproduce the issue
  
  Full model pytest on tt-xla (requires the maxpool shape-contract fix to even reach the failing comparison):
  
`  pytest -svv tests/runner/test_models.py::test_all_models_torch[maptr/pytorch-Nano_R18_110e-single_device-inference]`

  Isolated LayerNorm sanity (single torch.nn.LayerNorm on a transformer-stddev input):

`  pytest -svv maptr_layernorm_sanity.py `

[may14_layernorm_sanity.py](https://github.com/user-attachments/files/28150927/may14_layernorm_sanity.py)

  Platform

  - Wormhole n150 (via tt-xla)

**Comment by saiarthiraguram** [comment]
Issue no longer relevant, will reopen if necessary
